internal/executor/utils.go

```diff
@@ -100,51 +91,25 @@ func ShowDerivation(ctx context.Context, flakeUrl, hostname string) (drvPath str
 	return
 }
 
-type Path struct {
-	Path string `json:"path"`
-}
-
-type Output struct {
-	Out Path `json:"out"`
-}
-
-type Derivation struct {
-	Outputs Output `json:"outputs"`
-}
-
-type Show struct {
-	NixosConfigurations map[string]struct{} `json:"nixosConfigurations"`
-}
-
-func List(flakeUrl string) (hosts []string, err error) {
+func build(ctx context.Context, drvPath string) (err error) {
 	args := []string{
-		"flake",
-		"show",
-		"--json",
-		flakeUrl,
-	}
-	var stdout bytes.Buffer
-	err = runNixCommand(args, &stdout, os.Stderr)
-	if err != nil {
-		return
-	}
-
-	var output Show
-	err = json.Unmarshal(stdout.Bytes(), &output)
+		"build",
+		fmt.Sprintf("%s^*", drvPath),
+		"-L",
+		"--no-link"}
+	err = runNixCommand(args, os.Stdout, os.Stderr)
 	if err != nil {
 		return
 	}
-	hosts = make([]string, 0, len(output.NixosConfigurations))
-	for key := range output.NixosConfigurations {
-		hosts = append(hosts, key)
-	}
 	return
 }
 
-func Build(ctx context.Context, drvPath string) (err error) {
+// fetchBuild fetches the build output from the Nix binary cache
+// instead of evaluating locally.
+func fetchBuild(ctx context.Context, outPath string) (err error) {
```

Comment:
Since it is not currently used, could you remove it (and ideally moving it to a PR adding the Garnix executor)?